### DISCUSSION TITLE
graphql/inspector/docs/essentials/similar

### DISCUSSION THREAD

**Discussion by giscus** [kamilkisiela/graphql-inspector#2378] [Docs Discussions]
# graphql/inspector/docs/essentials/similar

GraphQL Inspector Documentation

https://the-guild.dev/graphql/inspector/docs/essentials/similar

**Comment by istominaver** [comment]
I think similar works somehow strange right now. It detects as similar 2 totally different types with no common fields or similarity in the names but they have the same amount of fields, but it doesn't detect 2 types which have some common/similar field names and similar type names but have different amount of fields.

For example, for next types:

type MyListingInfo {
    title: String, 
    price: String, 
    image: String
} 

type Listing {
    id: ID!, 
    title: String, 
    currentPrice: String,  
    originalRetailPrice: String
    image: String,
    fieldX: Boolean!, 
    fieldY: String, 
    fieldZ: String
}

type Cat {
    id: ID, 
    type: String, 
    test: String
} 

similar function detects MyListingInfo and Cat as similar, but does not detect MyListingInfo and Listing to be similar
